src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/GroupCollection.cs

```diff
@@ -210,12 +211,12 @@ void IList.RemoveAt(int index)
             return new Enumerator(this);
         }
 
-        public bool TryGetValue(string key, out Group value)
+        public bool TryGetValue(string key, [MaybeNullWhen(false)] out Group value)
```

Comment:
Same comment as in the ref: I think this should be `[NotNullWhen(true)] out Group? value`.